src/Generator/Generators/CLI/CLIMarshal.cs

```diff
@@ -40,13 +40,15 @@ public override bool VisitTagType(TagType tag, TypeQualifiers quals)
             return decl.Visit(this);
         }
 
+        private static int arrayNum = 0;
```

Comment:
I believe I've already sent you a remark about this. A static value keeps increasing for the lifetime of the type - that is, for the whole run of the application. This means that for each new array ANYWHERE you'll increase this index.
We already have a mechanism to deal with this. Context has properties called ParameterIndex and MarshalVarPrefix. See how they are used and employ them in your code.